csrc/lower_bank_conflict.cpp

```diff
@@ -79,19 +79,69 @@ inline int64_t getPhaseSize(int64_t word_size_bytes) {
   return 32;
 }
 
+// Doc for ldmatrix can be found at:
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#warp-level-matrix-load-instruction-ldmatrix
+//
+// According to this doc, ldmatrix always do vectorize 8 load, which might be
+// different from the consumer vectorization factor. The vectorization factor of
+// the ldmatrix consumer is not the load vector size (which is always 8),
+// instead, it is the number of items each thread will eventually get. This
+// vectorization factor determines the .num modifier of ldmatrix, which further
+// determines the number of addresses used by ldmatrix. For ldmatrix.x1, there
+// are 8 addresses. For ldmatrix.x2, there are 16 addresses. For ldmatrix.x4,
+// there are 32 addresses.
+int64_t getLdMatrixNumThreads(int64_t word_size) {
+  switch (word_size) {
+    case 2:
+      // If the consumer has vector 2, then each thread of the warp get 2 items.
+      // So there are in total 32*2 = 64 items for the warp. Each vector has 8
+      // elements, so there are 64/8 = 8 vectors, i.e. this is a .x1 ldmatrix
+      // and the first 8 threads contain useful addresses.
+      return 8;
+    case 4:
+      // If the consumer has vector 4, then each thread of the warp get 4 items.
+      // So there are in total 32*4 = 128 items for the warp. Each vector has 8
+      // elements, so there are 128/8 = 16 vectors, i.e. this is a .x2 ldmatrix
+      // and the first 16 threads contain useful addresses.
+      return 16;
+    case 8:
+      // If the consumer has vector 8, then each thread of the warp get 8 items.
+      // So there are in total 32*8 = 256 items for the warp. Each vector has 8
+      // elements, so there are 256/8 = 32 vectors, i.e. this is a .x4 ldmatrix
+      // and all the 32 threads contain useful addresses.
+      return 32;
+    default:
+      TORCH_INTERNAL_ASSERT(false, "Invalid word size for ldmatrix");
+  }
+}
+
 std::vector<int64_t> evaluateAddressesOnFirstPhase(
-    kir::TensorIndex* ti,
     const std::vector<kir::ForLoop*>& for_loops,
     c10::optional<LaunchParams> launch_params,
-    const ExpressionEvaluator& expr_eval_common) {
+    const ExpressionEvaluator& expr_eval_common,
+    LoadStoreOp* ldst,
+    bool is_producer) {
   std::vector<int64_t> addresses;
-  const auto word_size_bytes =
-      dataTypeSize(*(ti->getDataType())) * getVectorizeSize(ti);
```

Comment:
This is a bug, it shouldn't be the vectorize size of this tensor, but should always be the vectorize size of the consumer. If this tensor is consumer, then it is correct. But if it is a producer, then this will be wrong.